components/org.wso2.bfsi.consent.management.extensions/src/main/java/org/wso2/bfsi/consent/management/extensions/manage/impl/DefaultConsentManageHandler.java

```diff
@@ -0,0 +1,235 @@
+import org.wso2.bfsi.consent.management.extensions.manage.model.ConsentManageData;
+import org.wso2.bfsi.consent.management.extensions.manage.model.ConsentPayloadValidationResult;
+import org.wso2.bfsi.consent.management.extensions.manage.utils.ConsentManageConstants;
+import org.wso2.bfsi.consent.management.service.ConsentCoreService;
+
+
+/**
+ * Consent manage handler default implementation.
+ */
+public class DefaultConsentManageHandler implements ConsentManageHandler {
+
+    private static final Log log = LogFactory.getLog(DefaultConsentManageHandler.class);
+    private static final ConsentCoreService consentCoreService = ConsentExtensionsDataHolder.getInstance()
+            .getConsentCoreService();
+    @Override
+    public void handleGet(ConsentManageData consentManageData) throws ConsentException {
+
+        //Check whether client ID exists
+        if (StringUtils.isEmpty(consentManageData.getClientId())) {
+            log.error("Client ID missing in the request.");
+            throw new ConsentException(ResponseStatus.BAD_REQUEST, "Client ID missing in the request.");
+        }
+
+        if (consentManageData.getRequestPath() == null) {
+            throw new ConsentException(ResponseStatus.BAD_REQUEST, "Resource Path Not Found");
+        }
+        String consentId = consentManageData.getRequestPath().split("/")[1];
+        if (ConsentExtensionUtils.isConsentIdValid(consentId)) {
+            try {
+                ConsentResource consent = consentCoreService.getConsent(consentId, false);
+                if (consent == null) {
+                    throw new ConsentException(ResponseStatus.BAD_REQUEST, "Consent not found");
+                }
+                String consentType = ConsentExtensionUtils.getConsentType(consentManageData.getRequestPath());
+                if (!consentType.equals(consent.getConsentType())) {
+                    throw new ConsentException(ResponseStatus.BAD_REQUEST, "Consent Type mismatch");
+                }
+                // Check whether the client id is matching
+                if (!consent.getClientID().equals(consentManageData.getClientId())) {
+                    //Throwing same error as null scenario since client will not be able to identify if consent
+                    // exists if consent does not belong to them
+                    throw new ConsentException(ResponseStatus.BAD_REQUEST, "Client ID mismatch");
+                }
+                JSONObject receiptJSON = (JSONObject) new JSONParser(JSONParser.MODE_PERMISSIVE)
+                        .parse(consent.getReceipt());;
+                consentManageData.setResponsePayload(ConsentExtensionUtils.getInitiationRetrievalResponse(receiptJSON,
+                        consent));
+                consentManageData.setResponseStatus(ResponseStatus.OK);
+            } catch (ConsentManagementException | ParseException e) {
+                throw new ConsentException(ResponseStatus.INTERNAL_SERVER_ERROR,
```

Comment:
Let's log the exception

Reply:
Fixed in https://github.com/wso2-extensions/bfsi-consent-management/pull/4/commits/98992be82ffa4483d8c6f51e4dce81ee11bb9d3b